### ISSUE TITLE
Missing Units

### ISSUE THREAD

**Issue by Nick2253** [thomasloven/lovelace-template-entity-row#129]
When I use `template-entity-row`, the unit is missing from the entity state:

```
type: entities
title: Missing Units
entities:
  - type: custom:template-entity-row
    entity: >-
      {{ state_attr('sensor.person_data', 'data')['person.nick']['battery']
      }}
  - type: custom:template-entity-row
    entity: sensor.pixel_6_pro_battery_level
  - entity: sensor.pixel_6_pro_battery_level
```

![image](https://github.com/thomasloven/lovelace-template-entity-row/assets/926554/ddb2068a-791a-4774-9719-c8e91be00975)

You can see that the first row, which is a fully templated entity pulled out of my `sensor.person_data` data storage, and the second row, which directly refers to the `sensor.pixel_6_pro_battery_level`, lack the "%" unit symbol.  However, a normal entity does show the units.

**Comment by ildar170975** [comment]
>  unit is missing

As expected since the result is a STATE, not a state with a unit.
Compose a result using a unit if you need it.

**Comment by Nick2253** [comment]
> As expected since the result is a STATE, not a state with a unit.

I'm not sure why that would be "expected".  There no indication in the documentation that it behaves any differently than a normal entity row, so I think it's pretty logical to expect it to behave that way.  Since there is also a templateable `state` value, I'd think that I'd use that for anything that I'd want that's non-standard.

In any event, if this is indeed expected behavior, than consider this a feature request for a `default` or similar flag to display the entity row as if it were a default row.  Reading through the documentation again, the part about "blinds" stands out to me as another reason to want this kind of behavior.  And for more complicated entities like locks, thermostats, covers, etc, it would be great to be able to only template the entity, and leave everything else the same as the default entity row.

**Comment by ildar170975** [comment]
> There no indication in the documentation that it behaves any differently than a normal entity row, so I think it's pretty logical to expect it to behave that way

Then I would suggest you to prepare a PR for improving "readme" since `template-entity-row` DEFINITELY looks DIFFERENTLY.

![image](https://github.com/thomasloven/lovelace-template-entity-row/assets/71872483/13e5bcbf-2a2b-4e7a-aa57-15059ecac15a)

![image](https://github.com/thomasloven/lovelace-template-entity-row/assets/71872483/3f543fce-56cd-478a-8fdf-f34bd7ec6e56)

![image](https://github.com/thomasloven/lovelace-template-entity-row/assets/71872483/e6472abd-a148-41d0-9449-2fe06ea3329b)

In short - a state is always a "raw" & not localized state.

Alternatively - you may prepare a FR for displaying a "standard" row if some new option is set.

**Comment by Airblader** [comment]
From the point of view of "principle of least surprise", going from

```
- entity: sensor.foo
```

to

```
- entity: sensor.foo
  type: custom:template-entity-row
```

... I, too, would expect it to behave exactly the same, not having specified any of the new options template-entity-row provides.

Anyway, it's too late to make this work _by default_ now without breaking people. For anyone seeing this wanting a solution:

```
- entity: sensor.foo
  type: custom-template-entity-row
  state: >
    {{ states(config.entity) }} {{ state_attr(config.entity, 'unit_of_measurement') }}
```

**Comment by sameleff** [comment]
@Airblader thanks for posting this! I'm struggling to get the units for a state template, for example, something like this:

```
type: entities
entities:
  - type: custom:template-entity-row
    icon: mdi:thermometer-check
    name: Temp Delta
    state: >-
      {{ '%0.1f' | format(states('sensor.sunroom_controller_temperature') | float(0)
      - states('input_number.sunroom_temperature_setpoint') | float(0)) | float(0) }}
```

I'm defining the value to display directly within the `state` block, so when I append the `{{ state_attr(config.entity, 'unit_of_measurement') }}` there is no reference in the config to refer to.

EDIT: Got it to work using the below workaround. I'm sure there's a better way, but you can simply refer to an existing entity that has been configured with the correct units within the `state_attr()` line, e.g.:

```
type: entities
entities:
  - type: custom:template-entity-row
    icon: mdi:thermometer-check
    name: Temp Delta
    state: >-
      {{ '%0.1f' | format(states('sensor.sunroom_controller_temperature') | float(0)
      - states('input_number.sunroom_temperature_setpoint') | float(0)) | float(0) }}
      {{ state_attr('sensor.sunroom_controller_temperature', 'unit_of_measurement') }}
```
